bin/collator/src/parachain/service.rs

```diff
@@ -1179,23 +1191,28 @@ where
             client.clone(),
         );
 
-        basic_aura::run::<Block, AuraPair, _, _, _, _, _, _, _>(basic_aura::Params {
+        aura::run::<Block, AuraPair, _, _, _, _, _, _, _, _, _>(AuraParams {
             create_inherent_data_providers: move |_, ()| async move { Ok(()) },
             block_import: parachain_block_import.clone(),
             para_client: client.clone(),
+            para_backend: backend,
             relay_client: relay_chain_interface.clone(),
+            code_hash_provider: move |block_hash| {
+                client
+                    .code_at(block_hash)
+                    .ok()
+                    .map(|c| ValidationCode::from(c).hash())
+            },
             sync_oracle: sync_oracle.clone(),
             keystore,
             collator_key,
             para_id,
             overseer_handle,
-            slot_duration,
             relay_chain_slot_duration: Duration::from_secs(6),
             proposer: cumulus_client_consensus_proposer::Proposer::new(proposer_factory),
             collator_service,
-            // We got around 500ms for proposing
-            authoring_duration: Duration::from_millis(500),
-            collation_request_receiver: Some(request_stream),
+            authoring_duration: Duration::from_millis(1500),
```

Comment:
The block limit is defined on runtime by weight limit. This duration here is the client time available to build the block. If not built within 1.5 secs then it will timeout (hence slot hardware). Normally it needs to match with runtime block limit but this is a transitory period until async backing is enabled. We need this out so when the time comes, all clients support async backing and we just enable it.

Reply:
Right, `weight limit` would be the limiter of the block size, assuming that the collator hardware is at least powerful as the benchmark machine.

Setting this value to `1500` now means that collators _allow_ themselves to take up to `1500 [ms]` seconds to produce a block worth of `500 [ms]` (hence the addition of HW spec check as you said).

I was about to suggest to make this into a CLI param, but I assume that would defeat the purpose of avoiding the client restart 🙂.

I think it's all clear now. 👍

Reply:
Client is still required to build block within 0.5s, if not it will miss timeframe of 2s build&gossip. This doesn't mean they've 1.5s to build the block. It's more a time so client don't continue building if this timeout is reached. After this timeout, it's pointless to continue.